src/main/java/com/hazelcast/jdbc/HazelcastJdbcPreparedStatement.java

```
@@ -0,0 +1,370 @@
+package com.hazelcast.jdbc;
+
+import java.io.InputStream;
+import java.io.Reader;
+import java.math.BigDecimal;
+import java.net.URL;
+import java.sql.Array;
+import java.sql.Blob;
+import java.sql.Clob;
+import java.sql.Connection;
+import java.sql.Date;
+import java.sql.NClob;
+import java.sql.ParameterMetaData;
+import java.sql.PreparedStatement;
+import java.sql.Ref;
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+import java.sql.RowId;
+import java.sql.SQLException;
+import java.sql.SQLXML;
+import java.sql.Time;
+import java.sql.Timestamp;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.List;
+
+class HazelcastJdbcPreparedStatement extends HazelcastJdbcStatement implements PreparedStatement {
+
+    private final List<Object> parameters = new ArrayList<>();
+    private final String sql;
+
+    HazelcastJdbcPreparedStatement(String sql, HazelcastJdbcClient client, Connection connection) {
+        super(client, connection);
+        this.sql = sql;
+    }
+
+    @Override
+    public ResultSet executeQuery() throws SQLException {
+        checkClosed();
```

The `executeQuery`/`executeUpdate` methods should throw an exception if the type of the result is not as expected.